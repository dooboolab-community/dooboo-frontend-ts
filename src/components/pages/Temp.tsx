import Button from '../uis/Button';
import {ReactElement} from 'react';
import styled from '@emotion/styled';
import {useNavigate} from 'react-router-dom';

const Container = styled.div`
  flex: 1;
  align-self: stretch;
  justify-content: center;
  align-items: center;
  background: ${(props) => props.theme.background};
  padding: 50px;
`;

function Temp(): ReactElement {
  const navigate = useNavigate();

  return (
    <Container>
      <Button onClick={(): void => navigate(-1)} text="back to tab page" />
    </Container>
  );
}

export default Temp;
