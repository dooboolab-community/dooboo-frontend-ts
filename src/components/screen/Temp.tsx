import React, { Component } from 'react';
import Button from '../shared/Button';
import styled from 'styled-components';

const Container = styled.div`
  display: flex;
  height: 100vh;
  justify-content: center;
  align-items: center;
  background-color: ${(props) => props.theme.mainColor};
  padding: 50px;
`;

interface IProps {
  history?: any;
}

function Temp(props: IProps) {
  const { history } = props;
  return (
    <Container>
      <Button
        onClick={() => history.goBack()}
        text='back to tab page'
      />
    </Container>
  );
}

export default Temp;
