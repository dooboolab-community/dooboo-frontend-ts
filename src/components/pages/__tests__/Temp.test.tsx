import * as renderer from 'react-test-renderer';

import {RenderResult, fireEvent, render} from '@testing-library/react';

import React from 'react';
import Temp from '../Temp';
import {createTestElement} from '../../../../test/testUtils';

const props = {};
const component = createTestElement(<Temp {...props} />);

describe('[Temp] render', () => {
  it('renders without crashing', () => {
    const rendered = renderer.create(component).toJSON();

    expect(rendered).toBeTruthy();
  });
});

describe('[Temp] Interaction', () => {
  let renderResult: RenderResult;

  beforeEach(() => {
    renderResult = render(component);
  });

  it('should simulate [onClick] when [btn] has been clicked', () => {
    const btnInstance = renderResult.getByText('back to tab page');

    fireEvent.click(btnInstance);
  });
});
