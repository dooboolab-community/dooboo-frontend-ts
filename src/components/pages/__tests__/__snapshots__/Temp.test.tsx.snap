// Vitest Snapshot v1

exports[`[Temp] render > renders without crashing 1`] = `
<div
  style={
    {
      "flex": 1,
      "textAlign": "center",
      "width": "100vw",
    }
  }
>
  <div
    className="css-10njcwo"
  >
    <button
      className="css-wnig0z"
      onClick={[Function]}
      style={
        {
          "display": "flex",
          "height": "60px",
        }
      }
    >
      <div
        className="css-qfy95"
      >
        <span
          className="css-6apmke"
        >
          back to tab page
        </span>
      </div>
    </button>
  </div>
</div>
`;
